### PR TITLE
[less]: allow export type and add v4 options

### PR DESCRIPTION
**Pull request by Gehbt**
## Description
This PR let less could export type in node16/nodenext.

example
```cjs
/**
 * @type {import('less').Options}
 */
```

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/less/index.d.ts [hunk 1]
```diff
@@ -293,12 +216,93 @@ interface LessStatic {
...
 
-    FileManager: typeof Less.FileManager;
-    PluginManager: typeof Less.PluginManager;
+    FileManager: typeof FileManager;
+    PluginManager: typeof PluginManager;
 }
 
-declare module "less" {
-    export = less;
+declare class PluginManager {
```
Comment (jakebailey) [inline types/less/index.d.ts:223]: Why did these classes move outside the namespace? It doesn't seem like it'd be required at all for the PR, and it's making the diff pretty hard to understand.

**Review by jakebailey — CHANGES_REQUESTED** [review]
The latest change is definitely in the wrong direction; the library's export object is not a class and shouldn't be defined as one.

I'm not sure why you need to do that merge.

The global is provided by `export as namespace` already...

**Comment by jakebailey** [comment]
Can you clarify the problem you're trying to solve so I can better suggest a solution?

**Comment by Gehbt** [comment]
I made the pr just for it. And I have no more idea. 😢
```js
/**
 * @type {import('less').Options}
 */
```

**Pull request closed without merging**